lms/djangoapps/grades/tests/test_new.py

```diff
@@ -604,37 +600,19 @@ def test_course_grade_logging(self):
             course_id=self.course.id,
             enabled_for_course=True
         ):
-            with patch('lms.djangoapps.grades.new.course_grade.log') as log_mock:
-                # the course grade has not been created, so we expect each grade to be created
-                log_statement = u''.join((
-                    u"compute_and_update, read_only: {0}, subsections read/created: {1}/{2}, blocks ",
-                    u"accessed: {3}, total graded subsections: {4}"
-                )).format(False, 0, 3, 3, 2)
-                self._create_course_grade_and_check_logging(
-                    grade_factory,
-                    log_mock.warning,
-                    log_statement
-                )
-                log_mock.reset_mock()
-
-                # the course grade has been created, so we expect to read it from the db
-                log_statement = u"load_persisted_grade"
-                self._create_course_grade_and_check_logging(
-                    grade_factory,
-                    log_mock.info,
-                    log_statement
-                )
-                log_mock.reset_mock()
-
-                # only problem submission, a subsection grade update triggers
-                # a course grade update
-                self.submit_question_answer(u'test_problem_1', {u'2_1': u'choice_choice_2'})
-                log_statement = u''.join((
-                    u"compute_and_update, read_only: {0}, subsections read/created: {1}/{2}, blocks ",
-                    u"accessed: {3}, total graded subsections: {4}"
-                )).format(False, 3, 0, 3, 2)
-                self._create_course_grade_and_check_logging(
-                    grade_factory,
-                    log_mock.warning,
-                    log_statement
-                )
+            with patch('lms.djangoapps.grades.new.course_grade_factory.log') as log_mock:
+                # returns Zero when no grade, with ASSUME_ZERO_GRADE_IF_ABSENT
+                with waffle().override(ASSUME_ZERO_GRADE_IF_ABSENT, active=True):
+                    self._create_course_grade_and_check_logging(grade_factory.create, log_mock, u'CreateZero')
+
+                # read, but not persisted
+                self._create_course_grade_and_check_logging(grade_factory.create, log_mock, u'Update')
+
+                # update and persist
+                self._create_course_grade_and_check_logging(grade_factory.update, log_mock, u'Update')
+
+                # read from persistence, using create
+                self._create_course_grade_and_check_logging(grade_factory.create, log_mock, u'Read')
+
+                # read from persistence, using read
+                self._create_course_grade_and_check_logging(grade_factory.read, log_mock, u'Read')
```

Comment:
I ❤️ how clean this ended up looking. Would it be worth it to test these independently (as `ddt.data`) instead of sequentially? I know the two `grade_factory.create` calls are different depending on context so this may be too hard.